psbt/src/global.rs

```diff
@@ -80,7 +80,8 @@ impl PSBTGlobal {
     pub(crate) fn set_tx(&mut self, tx: &LegacyTx) {
         let tx_ins: Vec<BitcoinTxIn> = tx.inputs().iter().map(|i| i.unsigned()).collect();
 
```

Comment:
Maybe something like:

```rust
if tx_ins.len() > 0 {
    ... rest of code ...
}
```

To prevent a new tx being created with no vins here